client/common/types.ts

```diff
@@ -20,3 +22,9 @@ export type StatCalculator = (
 export type SetCounter = {
   [key: string]: { count: number; set: item_set };
 };
+
+export type FilterAction =
+  | { type: 'SEARCH'; search: string }
+  | { type: 'MAX_LEVEL'; maxLevel: number }
+  | { type: 'STATS'; stats: Array<Stat> }
+  | { type: 'ITEM_TYPE_IDS'; itemTypeIds: Array<CheckboxValueType> };
```

Comment:
How often do you want to filter by type specifically? Feel like generalizing it a bit to filter by slot may seem better ux wise.